File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int tillMin=INT_MAX;
        int ans =0;
        
        for(int i=0;i<prices.size();i++){
            ans = max(prices[i]-tillMin,ans);
            tillMin=min(prices[i],tillMin);
        }
        
        return ans == INT_MAX ? 0 : ans;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
    var tillMin=Number. MAX_VALUE ;
        var ans =0;
        
        for(var i=0;i<prices.length;i++){
            ans = Math.max(prices[i]-tillMin,ans);
            tillMin=Math.min(prices[i],tillMin);
        }
        
        return ans === Number. MAX_VALUE ? 0 : ans;
    
};